.github/workflows/lint.yaml

```diff
@@ -0,0 +1,46 @@
+name: Linting
```

Comment:
Looking at how these workflows are displayed in the Actions view,  I think it makes sense to pull 'docs' and 'modules' into their own workflows so that when individual jobs fail it's easier to identify quickly what might've gone wrong. I think that also makes sense organizationally, since the modules spec isn't particularly related to Linux for example, and generating the docs isn't related to linting the Ruby code in the repo. I don't feel really strongly about this if there's a good reason to group them together, but I'm not sure what that reason would be.

Reply:
Created separate 'docs' and 'modules' workflows. Since the modules are tested on both Linux and Windows, there are two jobs for the 'modules' workflow... that makes me wonder if instead of 'Linux' and 'Windows' workflows we would have 'Fast*' and 'Slow*' workflows with Linux and Windows jobs...

* naming TBD